pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/JuniperShipKitRoute.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsm.route;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.model.notPepperKit.*;
+import org.broadinstitute.dsm.security.RequestHandler;
+import spark.Request;
+import spark.Response;
+
+public class JuniperShipKitRoute extends RequestHandler {
+
+    NotPepperKitCreationService kitCreationService;
+
+    public JuniperShipKitRoute(NotPepperKitCreationService kitCreationService) {
+        this.kitCreationService = kitCreationService;
+    }
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        return createNonPepperKit(request, response);
+    }
+
+    public KitResponse createNonPepperKit(Request request, Response response) {
+        ShipKitRequest shipKitRequest = new Gson().fromJson(request.body(), ShipKitRequest.class);
+        JuniperKitRequest juniperKitRequest = shipKitRequest.getJuniperKitRequest();
+        if (juniperKitRequest == null) {
+            response.status(400);
+            return new KitResponse("EMPTY_REQUEST", null);
+        }
+        if (StringUtils.isBlank(shipKitRequest.getJuniperStudyGUID())) {
+            response.status(400);
+            return new KitResponse("EMPTY_STUDY_NAME", null);
+        }
+        if (StringUtils.isBlank(shipKitRequest.getKitType())) {
+            response.status(400);
+            return new KitResponse("EMPTY_KIT_TYPE", null);
+        }
+        KitResponse kitResponse = this.kitCreationService.createNonPepperKit(juniperKitRequest, shipKitRequest.getJuniperStudyGUID(),
+                shipKitRequest.getKitType());
+        if (kitResponse instanceof KitStatus) {
```

Comment:
I see what you mean... 
The reason I made it like that is, when we are returning a `KitStatus` object we can still have errorMessages in it that are not particularly related to this request, like the fedex was out or the cost object didn't work, stuff like that. However, if we had a bad request, we are going to return just a KitResponse with an error message related to that request. I was trying to somehow differentiate based on that.

Reply:
At this point KitResponse is identical to KitStatus, but are you imagining that KitStatus will contain more information in the future? If so, who will act on that information?
One way to handle this is "You aren't gonna need it" (YAGNI)... add KitStatus later when it actually is different from KitResponse. But in any case it feels like KitStatus "has a" KitResponse (class composition) rather than KitStatus "is a" KitResponse.
What do you think?

Reply:
Oh yes, KitStatus will have every tiny detail about the kit. sent date, received date, dsm shipping id, label, who scanned it and when, basically everything. 

Juniper will get it when they call the status endpoint (coming soon :) ) but they want to get the same object for a successful ship kit call as well. I haven't implemented that yet so for the first draft they are ok with just getting a 200 and this needs more consideration and may change in future.

Currently to me, KitStatus _is_ a KitResponse. A KitResponse could be an error, or it could be all of these other informations